### ISSUE TITLE
Svelte Kit project renders blank page

### ISSUE THREAD

**Issue by oliversturm** [codesandbox/codesandbox-client#6593]
# 🐛 bug report

## Preflight Checklist

- [X] I have read the [Contributing Guidelines][contributing] for this project.
- [X] I agree to follow the [Code of Conduct][code_of_conduct] that this project
      adheres to.
- [X] I have searched the issue tracker for an issue that matches the one I want
      to file, without success.

## Description of the problem

I created a Svelte Kit project locally (using the standard command `npm init svelte@next cstest-sveltekit-1`) and pushed it to this public repo: https://github.com/oliversturm/cstest-sveltekit-1

I imported it into CSB and a fork was created. However, the preview only shows a blank page. 

I tried this in Firefox (98.0.2 (64-bit) on Arch Linux) and Chromium (Version 100.0.4896.60, same 64 bit Arch Linux), and the results are similar - blank preview. 

One difference is that Chromium displays a couple errors in the console panel:

```
Problems consuming cache 
Error: Failed to read the 'localStorage' property from 'Window': Access is denied for this document.
```

These errors are refreshed at least once a second, it was pretty difficult to copy the text between refreshes. Eventually I found that I could open the browser console and grab screenshots of the two issues. At the bottom there's also an extra POST problem that didn't appear in the console panel.

![image](https://user-images.githubusercontent.com/4730969/161299210-d09f0109-a70d-4af3-b43e-32a37cdfecaf.png)

![image](https://user-images.githubusercontent.com/4730969/161299274-3485c670-e547-4651-a873-74668aa21a7b.png)

(The same steps are not possible in Firefox - the console is always empty there.)

## How has this issue affected you? What are you trying to accomplish?

I think I should be able to run this project in CSB.

### To Reproduce

I believe if you import the public repo https://github.com/oliversturm/cstest-sveltekit-1 you should see the same thing.

### Your Environment

| Software         | Name/Version |
| ---------------- | ------------ |
| Сodesandbox      |  e334d85ce            |
| Browser          | Described above  |
| Operating System | Arch Linux             |

[code_of_conduct]: https://github.com/codesandbox/codesandbox-client/blob/master/CODE_OF_CONDUCT.md
[contributing]: https://github.com/codesandbox/codesandbox-client/blob/master/CONTRIBUTING.md

**Comment by JamesACS** [comment]
Hey @oliversturm - From what I can see your project doesn't have an index.html file. Which would be needed for the CodeSandbox preview to work

**Comment by oliversturm** [comment]
> doesn't have an index.html file. Which would be needed for the CodeSandbox preview to work

Why? The Svelte Kit project does not have (or need) a file called `index.html`, true. That does not prevent it from working locally - why should it not work in CSB? And what does any of this have to do with the crazy error messages about local storage?

**Comment by oliversturm** [comment]
@JamesACS I wasn't sure who you were :) Now I know, we can certainly keep it here to benefit everybody.

Now, `index.html` file - that's a fun thing to suggest, but the Svelte Kit template simply doesn't have one. So I'm not really following - where do you think it would come from? This project is started by running `npm run dev` and then there's a server running on port 3000 (which renders the file `src/app.html` as a frame, in case that matters). Why does the preview not simply connect to port 3000 and read what's there?

I certainly know nothing about the way CSB works internally, but I don't understand how this relates to an `index.html` file provided by the template (or not, in this case). Could I somehow create an `index.html` independently that would satisfy the requirements? Would CSB be happy if the `App.html` was called `index.html`? Would a link to that file make a difference?

**Comment by JamesACS** [comment]
No trouble! 

You may find it handy to check out our Svelte template, our template has index.html in the public folder with only this code:

`<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf8" />
    <meta name="viewport" content="width=device-width" />

    <title>Svelte app</title>

    <link rel="stylesheet" href="public/bundle.css" />
  </head>

  <body>
    <script src="bundle.js"></script>
  </body>
</html>
`

It's purely a quirk of our preview implementation where you'll need to add the file so the preview knows where to find the rest of the code.

**Comment by JamesACS** [comment]
Sorry, my paste wasn't working there. 

You may find it useful to look at our svelte template @oliversturm - it'll give you a good idea about how to build svelte apps within CodeSandbox. 

Our implementation has the index.html file in the public folder, which just points to the CSS and js files needed to start the rest of the app

**Comment by oliversturm** [comment]
Well, I understand all that - that's how basic Svelte applications are usually configured (as per the standard template published by that project). But with Svelte Kit the standard template has been changed to use `src/App.html` with a different syntax (`%svelte.head%`, `%svelte.body%`) to generate the starting HTML content during a build (check it out, it's [here in my sample repo](https://github.com/oliversturm/cstest-sveltekit-1/blob/main/src/app.html)).

Before I start playing around wildly, can you point me to a place where this startup mechanism is documented in detail? The way it is, I see many open questions.

* Where will the `index.html` be found? Does it need to be in a specific path?

* Which files should be referenced inside it? For instance, your Svelte template `public/index.html` has `public/bundle.css` and `bundle.js`:

```html
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf8" />
    <meta name="viewport" content="width=device-width" />

    <title>Svelte app</title>

    <link rel="stylesheet" href="public/bundle.css" />
  </head>

  <body>
    <script src="bundle.js"></script>
  </body>
</html>
```

However, both of these files don't exist in the project source code. I see in `rollup.config.js` that they will be created during a build (but the paths are not exactly the ones from the `index.html`):

```js
export default {
  input: "index.js",
  output: {
  ...
    file: "public/bundle.js"
  },
  plugins: [
    svelte({
  ...
      css: css => {
        css.write("public/bundle.css");
      }
    }),
  ...
```

I'm afraid it's quite unclear to me what exactly CSB expects to find inside that `index.html` file. Of course it's no trouble to create it on top of the existing project template, but I really wouldn't know how to make sure that CSB is happy with the content.

* Since the missing files are created by the build process, I assume that CSB runs `npm run dev` (or `npm run build`?) to set things up. Is that correct? But wouldn't it then also run a dev server as a result? Does it somehow prevent that dev server from starting up? I'm afraid I still don't get why this `index.html` is used, plus a few files that need to be generated first, instead of simply accessing the dev server that is needed anyway to create the missing files! Or does CSB somehow run a build process that does /not/ work through the normal `npm run dev` script?

Finally, a separate thought - since Svelte Kit apps also have a server-side component, perhaps it would be altogether better (and probably easier) to use a container sandbox?

**Comment by CompuIves** [comment]
Hey @oliversturm! You're right, I think in this case it would be best to use a container sandbox, since there's a server component to it. The svelte implementation that we have that runs in the browser hasn't been updated to support Sveltekit, so I think this sandbox won't be able to behave as expected in this sense.

You can force it to be a container sandbox by adding a `sandbox.config.json` to the root of the project, and setting:

```
{
  "template": "node"
}
```

With CodeSandbox Projects, we'll be able to also make this process much easier in the future!

**Comment by oliversturm** [comment]
Thank you @CompuIves , that makes sense - I've done that before, no worries.

Out of curiosity, is my understanding correct that the Svelte "template" setting basically results in some magic to build the project, and that is why `index.html` is required in the way that was described, and explains where the missing files come from without an explicit build step?

**Comment by CompuIves** [comment]
Yep, the template determines how our in-browser bundler will transpile the project. In Svelte's case, we're adding a transpiler that can convert `.svelte` files to its corresponding JS (similar to Vue). I'm not sure anymore whether an `index.html` is required, but the biggest change is that we support `.svelte` files which will probably mount to the html file then.

**Comment by oliversturm** [comment]
Understood, thanks. I never looked into the inner workings of the non-container sandboxes, so this is good to know. I hope Svelte Kit support will arrive at some point - of course the server-side components may make it hard to support without a backend container, but it would be good to be able to import a project generated by the standard template without extra hassle.
